test/extended/prometheus/prometheus.go

```diff
@@ -49,6 +51,101 @@ type TelemeterClientConfig struct {
+
+		// TODO: remove the namespace when the bug is fixed.
+		namespacesToSkip = sets.New[string]("openshift-marketplace", // https://issues.redhat.com/browse/OCPBUGS-59763
+			"openshift-image-registry",               // https://issues.redhat.com/browse/OCPBUGS-59767
+			"openshift-operator-lifecycle-manager",   // https://issues.redhat.com/browse/OCPBUGS-59768
+			"openshift-cluster-samples-operator",     // https://issues.redhat.com/browse/OCPBUGS-59769
+			"openshift-cluster-version",              // https://issues.redhat.com/browse/OCPBUGS-57585
+			"openshift-cluster-csi-drivers",          // https://issues.redhat.com/browse/OCPBUGS-60159
+			"openshift-cluster-node-tuning-operator", // https://issues.redhat.com/browse/OCPBUGS-60258
+			"openshift-etcd",                         // https://issues.redhat.com/browse/OCPBUGS-60263
+		)
+	)
+
+	g.BeforeEach(func(ctx g.SpecContext) {
+		var err error
+
+		kubeClient, err := kubernetes.NewForConfig(oc.AdminConfig())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		nsExist, err := exutil.IsNamespaceExist(kubeClient, "openshift-monitoring")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if !nsExist {
+			g.Skip("openshift-monitoring namespace does not exist, skipping")
+		}
+
+		prometheusURL, err = helper.PrometheusRouteURL(ctx, oc)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Get public url of prometheus")
+		bearerToken, err = helper.RequestPrometheusServiceAccountAPIToken(ctx, oc)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Request prometheus service account API token")
+
+		if namespacesToSkip.Has(monitoringAuthTestNamespace) {
+			e2e.Logf("The namespace %s is not skipped because $MONITORING_AUTH_TEST_NAMESPACE is set to it", monitoringAuthTestNamespace)
+			namespacesToSkip.Delete(monitoringAuthTestNamespace)
+		}
+	})
+
+	g.It("should not be accessible without auth [Serial]", func() {
+		var errs []error
+
+		g.By("checking that targets reject the requests with 401 or 403")
+		execPod := exutil.CreateExecPodOrFail(oc.AdminKubeClient(), oc.Namespace(), "execpod-targets-authorization")
+		defer func() {
+			err := oc.AdminKubeClient().CoreV1().Pods(execPod.Namespace).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
+			o.Expect(err).NotTo(o.HaveOccurred(), fmt.Sprintf("Delete pod %s/%s", execPod.Namespace, execPod.Name))
+		}()
+
+		contents, err := helper.GetURLWithToken(helper.MustJoinUrlPath(prometheusURL, "api/v1/targets"), bearerToken)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		targets := &prometheusTargets{}
+		err = json.Unmarshal([]byte(contents), targets)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(len(targets.Data.ActiveTargets)).Should(o.BeNumerically(">=", 5))
+
+		expected := sets.New[int](http.StatusUnauthorized, http.StatusForbidden)
+		for _, target := range targets.Data.ActiveTargets {
+			ns := target.Labels["namespace"]
+			o.Expect(ns).NotTo(o.BeEmpty())
+			if monitoringAuthTestNamespace != "" && ns != monitoringAuthTestNamespace {
```

Comment:
nit: `monitoringAuthTestNamespace != ""` isn't needed given that `ns` cannot be empty.

Reply:
I (still) do not see how it works.

If `monitoringAuthTestNamespace != ""` is removed, then `ns != monitoringAuthTestNamespace` will be true if we do not set `monitoringAuthTestNamespace` by the env. var. `$MONITORING_AUTH_TEST_NAMESPACE`.
It leads to ignoring all non-empty namespaces.